heareval/predictions/task_predictions.py

```diff
@@ -954,19 +948,32 @@ def print_scores(mode, scores_and_trainers):
         batch_size=conf["batch_size"],
         in_memory=in_memory,
     )
+    best_trainer = best_grid_point.trainer
     # This hack is necessary because we use the best validation epoch to
     # choose the event postprocessing
-    best_trainer.fit_loop.current_epoch = best_model_epoch
+    best_trainer.fit_loop.current_epoch = best_grid_point.epoch
 
-    test_scores = best_trainer.test(
-        ckpt_path=best_model_path, test_dataloaders=test_dataloader
+    test_results = best_trainer.test(
+        ckpt_path=best_grid_point.model_path, test_dataloaders=test_dataloader
+    )
+    assert len(test_results) == 1, "Should have only one test dataloader"
+    test_results = test_results[0]
+
+    test_results.update(
+        {
+            "validation_score": best_grid_point.validation_score,
+            "hparams": best_grid_point.hparams,
+            "postprocessing": best_grid_point.postprocessing,
+            "epoch": best_grid_point.epoch,
+            "time_in_min": best_grid_point.time_in_min,
+            "score_mode": best_grid_point.score_mode,
+            "embedding_path": embedding_path,
+        }
     )
-    assert len(test_scores) == 1, "Should have only one test dataloader"
-    test_scores = test_scores[0]
-
     open(embedding_path.joinpath("test.predicted-scores.json"), "wt").write(
-        json.dumps(test_scores, indent=4)
+        json.dumps(test_results, indent=4)
```

Comment:
Something in here isn't json serializable

Reply:
looking now